### ISSUE TITLE
Anaconda error with import numpy error with --standalone option

### ISSUE THREAD

**Issue by hasemi** [Nuitka/Nuitka#3268]
When I convert a minimal program using numpy with --standalone option, I get an error and cannot run it.
When I do not use --standalone option, the converted exe program works.

- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

```
> python -m nuitka --version
2.5.6
Commercial: None
Python: 3.10.15 | packaged by conda-forge | (main, Oct 16 2024, 01:15:49) [MSC v.1941 64 bit (AMD64)]
Flavor: Anaconda Python
Executable: ~\miniconda3\envs\test\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 11
Version C compiler: C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.27.29110\bin\Hostx64\x64\cl.exe (cl.exe).
```

- How did you install Nuitka and Python

Anaconda (conda-forge)

- The specific PyPI names and versions

```
> conda list
Name                    Version                   Build  Channel
anyio                     3.7.1              pyhd8ed1ab_0    conda-forge
argon2-cffi               23.1.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0          py310ha8f682b_5    conda-forge
arrow                     1.3.0              pyhd8ed1ab_0    conda-forge
asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge
async-lru                 2.0.4              pyhd8ed1ab_0    conda-forge
attrs                     24.2.0             pyh71513ae_0    conda-forge
babel                     2.14.0             pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
beautifulsoup4            4.12.3             pyha770c72_0    conda-forge
bleach                    6.1.0              pyhd8ed1ab_0    conda-forge
brotli                    1.1.0                h2466b09_2    conda-forge
brotli-bin                1.1.0                h2466b09_2    conda-forge
brotli-python             1.1.0           py310h9e98ed7_2    conda-forge
bzip2                     1.0.8                h2466b09_7    conda-forge
c-compiler                1.8.0                hcfcfb64_1    conda-forge
ca-certificates           2024.12.14           h56e8100_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cairo                     1.18.0               h32b962e_3    conda-forge
certifi                   2024.12.14         pyhd8ed1ab_0    conda-forge
cffi                      1.17.1          py310ha8f682b_0    conda-forge
charset-normalizer        3.4.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
contourpy                 1.3.0           py310hc19bc0b_2    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
debugpy                   1.8.7           py310h9e98ed7_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
double-conversion         3.3.0                h63175ca_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
et_xmlfile                2.0.0              pyhd8ed1ab_0    conda-forge
exceptiongroup            1.2.2              pyhd8ed1ab_0    conda-forge
executing                 2.1.0              pyhd8ed1ab_0    conda-forge
expat                     2.6.3                he0c23c2_0    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_3    conda-forge
fontconfig                2.14.2               hbde0cde_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.54.1          py310h38315fa_1    conda-forge
fqdn                      1.5.1              pyhd8ed1ab_0    conda-forge
freetype                  2.12.1               hdaf720e_2    conda-forge
graphite2                 1.3.13            h63175ca_1003    conda-forge
h11                       0.14.0             pyhd8ed1ab_0    conda-forge
h2                        4.1.0              pyhd8ed1ab_0    conda-forge
harfbuzz                  9.0.0                h2bedf89_1    conda-forge
hpack                     4.0.0              pyh9f0ad1d_0    conda-forge
httpcore                  1.0.6              pyhd8ed1ab_0    conda-forge
httpx                     0.27.2             pyhd8ed1ab_0    conda-forge
hyperframe                6.0.1              pyhd8ed1ab_0    conda-forge
icu                       75.1                 he0c23c2_0    conda-forge
idna                      3.10               pyhd8ed1ab_0    conda-forge
importlib-metadata        8.5.0              pyha770c72_0    conda-forge
importlib_metadata        8.5.0                hd8ed1ab_0    conda-forge
importlib_resources       6.4.5              pyhd8ed1ab_0    conda-forge
intel-openmp              2024.2.1          h57928b3_1083    conda-forge
ipykernel                 6.14.0          py310hbbfc1a7_0    conda-forge
ipython                   8.4.0           py310h5588dad_0    conda-forge
isoduration               20.11.0            pyhd8ed1ab_0    conda-forge
jedi                      0.19.1             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.4              pyhd8ed1ab_0    conda-forge
joblib                    1.4.2              pyhd8ed1ab_0    conda-forge
json5                     0.9.25             pyhd8ed1ab_0    conda-forge
jsonpointer               3.0.0           py310h5588dad_1    conda-forge
jsonschema                4.23.0             pyhd8ed1ab_0    conda-forge
jsonschema-specifications 2024.10.1          pyhd8ed1ab_0    conda-forge
jsonschema-with-format-nongpl 4.23.0               hd8ed1ab_0    conda-forge
jupyter-lsp               2.2.5              pyhd8ed1ab_0    conda-forge
jupyter_client            8.6.3              pyhd8ed1ab_0    conda-forge
jupyter_core              5.7.2           py310h5588dad_0    conda-forge
jupyter_events            0.10.0             pyhd8ed1ab_0    conda-forge
jupyter_server            2.7.0              pyhd8ed1ab_0    conda-forge
jupyter_server_terminals  0.5.3              pyhd8ed1ab_0    conda-forge
jupyterlab                4.2.5              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.3.0              pyhd8ed1ab_1    conda-forge
jupyterlab_server         2.27.3             pyhd8ed1ab_0    conda-forge
kiwisolver                1.4.7           py310hc19bc0b_0    conda-forge
krb5                      1.21.3               hdf4eb48_0    conda-forge
lcms2                     2.16                 h67d730c_0    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libblas                   3.9.0              25_win64_mkl    conda-forge
libbrotlicommon           1.1.0                h2466b09_2    conda-forge
libbrotlidec              1.1.0                h2466b09_2    conda-forge
libbrotlienc              1.1.0                h2466b09_2    conda-forge
libcblas                  3.9.0              25_win64_mkl    conda-forge
libclang13                19.1.2          default_ha5278ca_1    conda-forge
libdeflate                1.22                 h2466b09_0    conda-forge
libexpat                  2.6.3                he0c23c2_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libglib                   2.82.2               h7025463_0    conda-forge
libhwloc                  2.11.1          default_h8125262_1000    conda-forge
libiconv                  1.17                 hcfcfb64_2    conda-forge
libintl                   0.22.5               h5728263_3    conda-forge
libjpeg-turbo             3.0.0                hcfcfb64_1    conda-forge
liblapack                 3.9.0              25_win64_mkl    conda-forge
libpng                    1.6.44               h3ca93ac_0    conda-forge
libsodium                 1.0.20               hc70643c_0    conda-forge
libsqlite                 3.47.0               h2466b09_1    conda-forge
libtiff                   4.7.0                hfc51747_1    conda-forge
libwebp-base              1.4.0                hcfcfb64_0    conda-forge
libxcb                    1.16                 h013a479_1    conda-forge
libxml2                   2.12.7               h0f24e4e_4    conda-forge
libxslt                   1.1.39               h3df6e99_0    conda-forge
libzlib                   1.3.1                h2466b09_2    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markupsafe                3.0.2           py310h38315fa_0    conda-forge
matplotlib                3.9.2           py310h5588dad_1    conda-forge
matplotlib-base           3.9.2           py310h37e0a56_1    conda-forge
matplotlib-inline         0.1.7              pyhd8ed1ab_0    conda-forge
mistune                   3.0.2              pyhd8ed1ab_0    conda-forge
mkl                       2024.2.2            h66d3029_14    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
nbclient                  0.10.0             pyhd8ed1ab_0    conda-forge
nbconvert-core            7.16.4             pyhd8ed1ab_1    conda-forge
nbformat                  5.10.4             pyhd8ed1ab_0    conda-forge
nest-asyncio              1.6.0              pyhd8ed1ab_0    conda-forge
notebook-shim             0.2.4              pyhd8ed1ab_0    conda-forge
nuitka                    2.5.6           py310h8d17308_0    conda-forge
numpy                     2.1.2           py310hb9d903e_0    conda-forge
openjpeg                  2.5.2                h3d672ee_0    conda-forge
openpyxl                  3.1.5           py310h8d17308_1    conda-forge
openssl                   3.4.0                h2466b09_0    conda-forge
ordered-set               4.1.0              pyhd8ed1ab_0    conda-forge
overrides                 7.7.0              pyhd8ed1ab_0    conda-forge
packaging                 24.1               pyhd8ed1ab_0    conda-forge
pandas                    2.2.3           py310hb4db72f_1    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.4              pyhd8ed1ab_0    conda-forge
patsy                     0.5.6              pyhd8ed1ab_0    conda-forge
pcre2                     10.44                h3d7b363_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    10.4.0          py310h3e38d90_1    conda-forge
pip                       24.3.1             pyh8b19718_0    conda-forge
pixman                    0.43.4               h63175ca_0    conda-forge
pkgutil-resolve-name      1.3.10             pyhd8ed1ab_1    conda-forge
platformdirs              4.3.6              pyhd8ed1ab_0    conda-forge
prometheus_client         0.21.0             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.48             pyha770c72_0    conda-forge
psutil                    6.1.0           py310ha8f682b_0    conda-forge
pthread-stubs             0.4               hcd874cb_1001    conda-forge
pthreads-win32            2.9.1                h2466b09_4    conda-forge
pure_eval                 0.2.3              pyhd8ed1ab_0    conda-forge
pycparser                 2.22               pyhd8ed1ab_0    conda-forge
pygments                  2.18.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.2.0              pyhd8ed1ab_1    conda-forge
pyside6                   6.8.0.2         py310h60c6385_0    conda-forge
pysocks                   1.7.1           py310h5588dad_5    conda-forge
python                    3.10.15         hfaddaf0_2_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-fastjsonschema     2.20.0             pyhd8ed1ab_0    conda-forge
python-json-logger        2.0.7              pyhd8ed1ab_0    conda-forge
python-tzdata             2024.2             pyhd8ed1ab_0    conda-forge
python_abi                3.10                    5_cp310    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
pywin32                   307             py310h9e98ed7_3    conda-forge
pywinpty                  2.0.14          py310h9e98ed7_0    conda-forge
pyyaml                    6.0.2           py310ha8f682b_1    conda-forge
pyzmq                     26.2.0          py310h656833d_3    conda-forge
qhull                     2020.2               hc790b64_5    conda-forge
qt6-main                  6.8.0                hfb098fa_0    conda-forge
referencing               0.35.1             pyhd8ed1ab_0    conda-forge
requests                  2.32.3             pyhd8ed1ab_0    conda-forge
rfc3339-validator         0.1.4              pyhd8ed1ab_0    conda-forge
rfc3986-validator         0.1.1              pyh9f0ad1d_0    conda-forge
rpds-py                   0.20.0          py310hc226416_1    conda-forge
scikit-learn              1.5.2           py310hf2a6c47_1    conda-forge
scipy                     1.14.1          py310hbd0dde3_1    conda-forge
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                75.1.0             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sniffio                   1.3.1              pyhd8ed1ab_0    conda-forge
soupsieve                 2.5                pyhd8ed1ab_1    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
statsmodels               0.14.4          py310hb0944cc_0    conda-forge
tbb                       2021.13.0            hc790b64_0    conda-forge
terminado                 0.15.0          py310h5588dad_0    conda-forge
threadpoolctl             3.5.0              pyhc1e730c_0    conda-forge
tinycss2                  1.4.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.13               h5226925_1    conda-forge
tomli                     2.0.2              pyhd8ed1ab_0    conda-forge
tornado                   6.4.1           py310ha8f682b_1    conda-forge
traitlets                 5.14.3             pyhd8ed1ab_0    conda-forge
types-python-dateutil     2.9.0.20241003     pyhff2d567_0    conda-forge
typing-extensions         4.12.2               hd8ed1ab_0    conda-forge
typing_extensions         4.12.2             pyha770c72_0    conda-forge
typing_utils              0.1.0              pyhd8ed1ab_0    conda-forge
tzdata                    2024b                hc8b5060_0    conda-forge
ucrt                      10.0.22621.0         h57928b3_1    conda-forge
unicodedata2              15.1.0          py310ha8f682b_1    conda-forge
uri-template              1.3.0              pyhd8ed1ab_0    conda-forge
urllib3                   2.2.3              pyhd8ed1ab_0    conda-forge
vc                        14.3                ha32ba9b_22    conda-forge
vc14_runtime              14.40.33810         hcc2c482_22    conda-forge
vs2015_runtime            14.40.33810         h3bf8584_22    conda-forge
vs2019_win-64             19.29.30139         he1865b1_23    conda-forge
vswhere                   3.1.7                h57928b3_0    conda-forge
wcwidth                   0.2.13             pyhd8ed1ab_0    conda-forge
webcolors                 24.8.0             pyhd8ed1ab_0    conda-forge
webencodings              0.5.1              pyhd8ed1ab_2    conda-forge
websocket-client          1.8.0              pyhd8ed1ab_0    conda-forge
wheel                     0.44.0             pyhd8ed1ab_0    conda-forge
win_inet_pton             1.1.0           py310h5588dad_5    conda-forge
winpty                    0.4.3                         4    conda-forge
xorg-libxau               1.0.11               hcd874cb_0    conda-forge
xorg-libxdmcp             1.1.3                hcd874cb_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
yaml                      0.2.5                h8ffe710_2    conda-forge
zeromq                    4.3.5                ha9f60a1_6    conda-forge
zipp                      3.20.2             pyhd8ed1ab_0    conda-forge
zlib                      1.3.1                h2466b09_2    conda-forge
zstandard                 0.23.0          py310he5e10e1_1    conda-forge
zstd                      1.5.6                h0ea2cb4_0    conda-forge
```

- Many times when you get an error from Nuitka, your setup may be special

When I convert a program that uses tkinter with plugin tk-inter, the converted exe program works perfectly.

- Also supply a Short, Self Contained, Correct, Example

Minimal program which cannot be converted correctly is here.

- script.py

```
import numpy as np

a = np.linspace(0, 10, 50)
print(a)
```

- error message

```
> script.exe
Traceback (most recent call last):
  File "C:\Users\KPITC\DOCUME~1\test\exe\test\temp\SCRIPT~1.DIS\numpy\_core\__init__.py", line 23, in <module numpy._core>
  File "C:\Users\KPITC\DOCUME~1\test\exe\test\temp\SCRIPT~1.DIS\numpy\_core\multiarray.py", line 10, in <module numpy._core.multiarray>
  File "C:\Users\KPITC\DOCUME~1\test\exe\test\temp\SCRIPT~1.DIS\numpy\_core\overrides.py", line 8, in <module numpy._core.overrides>
ImportError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\KPITC\DOCUME~1\test\exe\test\temp\SCRIPT~1.DIS\numpy\__init__.py", line 114, in <module numpy>
  File "C:\Users\KPITC\DOCUME~1\test\exe\test\temp\SCRIPT~1.DIS\numpy\__config__.py", line 4, in <module numpy.__config__>
  File "C:\Users\KPITC\DOCUME~1\test\exe\test\temp\SCRIPT~1.DIS\numpy\_core\__init__.py", line 49, in <module numpy._core>
ImportError:

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.10 from "C:\Users\KPITC\DOCUME~1\test\exe\test\temp\SCRIPT~1.DIS\python.exe"
  * The NumPy version is: "2.1.2"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was:


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\KPITC\DOCUME~1\test\exe\test\temp\SCRIPT~1.DIS\script.py", line 1, in <module>
  File "C:\Users\KPITC\DOCUME~1\test\exe\test\temp\SCRIPT~1.DIS\numpy\__init__.py", line 119, in <module numpy>
ImportError: Error importing numpy: you should not try to import numpy from
        its source directory; please exit the numpy source tree, and relaunch
        your python interpreter from there.
```

- Command I use to convert

```
> nuitka --standalone script.py
Nuitka-Options: Used command line options: --standalone script.py
Nuitka: Starting Python compilation with Nuitka '2.5.6' on Python '3.10' commercial grade 'not installed'.
Nuitka: Completed Python level compilation and optimization.
Nuitka: Generating source code for C backend compiler.
Nuitka: Running data composer tool for optimal constant value handling.
Nuitka: Running C compilation via Scons.
Nuitka-Scons: Backend C compiler: cl.exe (cl.exe).
Nuitka-Scons: Backend linking program with 147 files (no progress information available for this stage).
Nuitka-Scons: Compiled 147 C files using clcache with 147 cache hits and 0 cache misses.
Nuitka-Plugins:dll-files: Found 24 files DLLs from numpy installation.
Nuitka: Keeping build directory 'script.build'.
Nuitka: Successfully created 'script.dist\script.exe'.
```

Thank you in advance.

**Comment by kayhayen** [comment]
For Anaconda, there is a fix on the factory branch that will be in 2.5.8, and I expect this to be the same issue #3261

Thanks for your report. This is fixed on the factory branch, which is a development version that is under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback on whether this is working is very welcome. Please do not share plans to do it; only confirm or deny that it is working.

**Comment by kayhayen** [comment]
This is fixed in 2.5.8 which saw a mini sprint, where I added Anaconda coverage on Nuitka-Watch, largely for Windows, but also with opencv (and hence numpy) on all 3 platforms. I saw that a generic warning is given about unnecessary configuration for numpy, that will be removed at a later date.

This should prevent Anaconda from regressing again in the future.
